storage/innobase/include/ut0new.h

```diff
@@ -896,6 +968,18 @@ ut_delete_array(
 
 #define ut_free(ptr)			::free(ptr)
 
+static inline void ut_free_dodump(void *ptr, ut_new_pfx_t *key)
+{
+	ulint size= key->m_size;
+
+#ifdef UNIV_PFS_MEMORY
+	deallocate_trace(key);
+#endif
+	ut_dodump(ptr, key);
+
+	os_mem_free_large(ptr, size);
```

Comment:
Why do we need to assign `size = key->m_size`? If `deallocate_trace(key)` invalidates `key->m_size`, then already the call `ut_dodump(ptr, key)` would be using an invalid `key`.